code/modules/projectiles/guns/projectile/sniper.dm

```diff
@@ -20,6 +20,12 @@
 	slot_flags = SLOT_BACK
 	actions_types = list()
 
+/obj/item/gun/projectile/automatic/sniper_rifle/process_fire(atom/target, mob/living/user, message = TRUE, params, zone_override, bonus_spread = 0)
+	if(istype(chambered.BB, /obj/item/projectile/bullet/sniper/antimatter) && !zoomed)
+		to_chat(user, "<span class='warning'>[src] must be zoomed in to fire this ammunition accurately!</span>")
+		bonus_spread += 60
```

Comment:
Maybe it would be better as a var on sniper bullets, and then typecast and apply it?
i.e.
```
/obj/item/projectile/bullet/sniper
	var/non_zoom_spread = 0
	
/obj/item/projectile/bullet/sniper/antimatter
	non_zoom_spread = 60
```

Reply:
yes, please do this @Qwertytoforty

Reply:
done!